### PR TITLE
test_scrub_log.py: Added test to verify the scrub & deep-scrub logs

### PR DESCRIPTION
**Pull request by pdhiran**
1. create 1 replicated and 1 EC pools, fill some data.
2. Initiate scrub & deep-scrub on the pools.
3. Verify OSD logs for the same

Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

# Description

Source of Test case - Close loop of customer BZs

### CODE REVIEW

**Comment by pdhiran** [comment]
Pass logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-EXG2SC

**Comment by pdhiran** [comment]
Pass logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-8PBHK4

Made some changes after comment from Prashant:
``` Prashant Dhange, Fri 11:03 PM, Edited
The primary OSD initiates the scrub for this PG when instructed (either forcefully by pg scrub command or scrub has been scheduled for it). The primary OSD is the one who sends scrub requests to replica OSDs. The replica OSDs never initiates scrub op for a PG. The primary OSD asks replica OSDs from PG acting set to reserves the resource for scrubbing and once all replicas confirms reservation then scrubbing kick in. It's just like client IO where primary OSD gets client IO op which sends subops to replica OSDs to complete the op.
 ```

File: tests/rados/test_scrub_log.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        lines: log line that needs to be matched
+    Returns:  Pass -> True, Fail -> False
+    """
+    log.debug("Checking if the log messages are generated in the OSD")
+    log_lines = rados_obj.get_journalctl_log(
+        start_time=start_time,
+        end_time=end_time,
+        daemon_type="osd",
+        daemon_id=osd,
+    )
```
Comment (sunilkumarn417) [inline tests/rados/test_scrub_log.py:171]: add debug log record for log_lines

Comment (pdhiran) [reply tests/rados/test_scrub_log.py:171]: done

Comment (pdhiran) [reply tests/rados/test_scrub_log.py:171]: done

**Comment by pdhiran** [comment]
Pass logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-4A3YDA

**Comment by pdhiran** [comment]
Pass logs :
 http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-N01NI4 
 http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-Q2IBSB

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**